systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaOauthExampleClients.java

```diff
@@ -91,11 +119,19 @@ public KafkaOauthExampleClients build() {
 
     protected KafkaOauthExampleClients(KafkaOauthExampleClients.Builder builder) {
         super(builder);
-        if (builder.oauthClientId == null || builder.oauthClientId.isEmpty()) throw new InvalidParameterException("OAuth client id is not set.");
-        if (builder.oauthClientSecret == null || builder.oauthClientSecret.isEmpty()) throw new InvalidParameterException("OAuth client secret is not set.");
+        if ((builder.oauthClientId == null || builder.oauthClientId.isEmpty()) &&
+                (builder.oauthConsumerClientId == null && builder.oauthProducerClientId == null)) throw new InvalidParameterException("OAuth clientId is not set.");
+        if (builder.oauthClientSecret == null || builder.oauthClientSecret.isEmpty() &&
+                (builder.oauthProducerSecret == null && builder.oauthConsumerSecret == null)) throw new InvalidParameterException("OAuth client secret is not set.");
+
```

Comment:
We discussed offline this -> I was thinking on situation like - I want to create just producer and send messages, with `oauthProducerClientId` and `oauthProducerSecret` specified, I'll leave the `oauthClientId` and `oauthClientSecret` empty, should this throw the `InvalidParameterException`?

I guess it doesn't matter, because we are using both consumer and producer if we want to have different secret and client id, but it's just suggestion.

WDYT @Frawless @see-quick

Reply:
What I proposed/suggested on this topic to @im-konge : 
- for "single/any" client with "defaults" use existing `oauthClientId` + `oauthClientSecret`
- in case you need to differentiate prod/cons, use those new pairs

I am open to suggestions. "the fix" was "just" to change 
`(builder.oauthConsumerClientId == null && builder.oauthProducerClientId == null)` ->
`(builder.oauthConsumerClientId == null || builder.oauthProducerClientId == null)` etc..